### PR TITLE
pubsub: provide a way to change number of streams

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java [hunk 1]
```diff
@@ -589,6 +589,15 @@ public Builder setSystemExecutorProvider(ExecutorProvider executorProvider) {
       return this;
     }
 
+    /**
+     * Sets the number of pullers used to pull messages from the subscription. Defaults to the
+     * number of available processors.
+     */
+    public Builder setNumPullers(int numPullers) {
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: This naming is a little opaque (for Java). Maybe: `setParallelPullCount` or something like that.

Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: Do you have preference between `setParallelPullCount` and `setNumStreams`? I like the latter for being a little shorter but that's not a very important metric.

Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: I think the first is more meaningful from the user's perspective - they don't have to know that pubsub is implemented with streams exactly. In Java, length is a less important metric than other languages :-P

**Comment by garrettjonesgoogle** [comment]
Side question - would this PR resolve #1683 ?

**Comment by pongad** [comment]
Yes it does!

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**